exchanges/binance/binance_websocket.go

```diff
@@ -39,54 +39,43 @@ func (b *Binance) WsConnect() error {
 		listenKey, err = b.GetWsAuthStreamKey()
 		if err != nil {
 			b.Websocket.SetCanUseAuthenticatedEndpoints(false)
-			log.Errorf(log.ExchangeSys, "%v unable to connect to authenticated Websocket. Error: %s", b.Name, err)
+			log.Errorf(log.ExchangeSys,
+				"%v unable to connect to authenticated Websocket. Error: %s",
+				b.Name,
+				err)
+		} else {
+			// cleans on failed connection
+			clean := strings.Split(b.Websocket.GetWebsocketURL(), "?streams=")
```

Comment:
When we have a URL change instigated via gctcli, we can't just apply our base URL because that will override what we set. And on failing to connect and thus retrying it would keep appending to the string URL string; this is a fix for that.